packages/store/src/defineHeapKind.js

```diff
@@ -0,0 +1,72 @@
+// @ts-check
+
+import { Far } from '@endo/marshal';
+
+const { ownKeys } = Reflect;
+
+const becomePassable = (tag, thing) => {
+  const names = ownKeys(thing);
+  if (names.length === 0 || typeof thing[names[0]] === 'function') {
+    return Far(tag, thing);
+  }
+  for (const name of names) {
+    assert.typeof(name, 'string');
+    Far(`${tag} ${name}`, thing[name]);
+  }
+  return harden(thing);
+};
+harden(becomePassable);
+
+/**
+ * Ersatz defineKind until virtual objects system is stable.
+ *
+ * @param {string} tag
+ * @param {Function} init
+ * @param {Function} actualize
+ * @param {Function} [finish]
+ */
+export const defineHeapKind = (tag, init, actualize, finish) => {
+  let nextInstanceID = 1;
+  const propertyNames = new Set();
+
+  function makeRepresentative(innerSelf) {
+    const wrappedData = {};
+    for (const prop of propertyNames) {
+      Object.defineProperty(wrappedData, prop, {
+        get: () => {
+          return innerSelf.rawData[prop];
+        },
+        set: value => {
+          innerSelf.rawData[prop] = value;
+        },
+      });
+    }
+    harden(wrappedData);
+
+    const representative = actualize(wrappedData);
+    return [representative, wrappedData];
+  }
+
+  function makeNewInstance(...args) {
+    const objID = `nonvirtual/${nextInstanceID}`;
+    nextInstanceID += 1;
```

Comment:
If you're totally faking it, then I think you can get rid of `objID` and `nextInstance` entirely.